packages/core/src/lib/base_protocol.ts

```diff
@@ -89,4 +93,15 @@ export class BaseProtocol implements IBaseProtocol {
     // Filter the peers based on the specified criteria
     return filterPeers(allPeersForProtocol, numPeers, maxBootstrapPeers);
   }
+
+  protected ensurePubsubTopicIsValid(
+    pubsubTopic: PubSubTopic,
+    allowedTopics: PubSubTopic[]
```

Comment:
Maybe _allowed_ is not the best terminology.
- initiated
- subscribed
- configured
Could be alternatives